lessons/module-3/react-3-advanced-data-management.md

```diff
@@ -226,121 +91,188 @@ Pair up and work together on implementing the functionality to `fetch` our ideas
 
 As we've noted before, we should use our catch somehow in order to display an error if something goes wrong.  Try stopping your backend server and see what happens.  Your catch should before firing (because the Promise rejected).  Let's do something with that error.
 
-Let's edit our intial state in our `App.js`:
+### Effect Hook
+
+The Effect Hook, `useEffect`, allows us to perform side effects (data fetching, subscriptions, etc) from a functional component.
+So far in React, we've leveraged the `useState` hook to manage our application data. Another common hook that you'll want to get comfortable with is `useEffect`. This hook allows you to perform some logic during certain phases of the **component lifecycle**. We'll first demonstrate how and when `useEffect` runs, and then we'll dig into the syntax before demonstrating some real world use-cases.
 
-```js
-// App.js
+### The Component Lifecycle
 
-this.state = {
-  ideas: [],
-  error: ''
-};
+Every component you create goes through several phases of existing:
+
+* **Mounting:** the component is being assessed, created and rendered on the DOM
```

Comment:
Might add clarification:

Every time the component updates, is it also re-mounting? Or does the "mounting" lifecycle only happen once, the first time the component is rendered?